krabs/krabs/ut.hpp

```diff
@@ -237,15 +237,18 @@ namespace krabs { namespace details {
 
         // for MOF providers, EventHeader.Provider is the *Message* GUID
         // we need to ask TDH for event information in order to determine the
-        // correct provider to pass this event to
-        auto schema = get_event_schema_from_tdh(record);
-        auto eventInfo = reinterpret_cast<PTRACE_EVENT_INFO>(schema.get());
-        for (auto& provider : trace.providers_) {
-            if (eventInfo->ProviderGuid == provider.get().guid_) {
-                provider.get().on_event(record, trace.context_);
-                return;
+        // correct provider to pass this event to. Don't throw in callback.
+        try
+        {
+            auto eventInfo = trace.context_.schema_locator.get_event_schema(record);
+            for (auto& provider : trace.providers_) {
+                if (eventInfo->ProviderGuid == provider.get().guid_) {
+                    provider.get().on_event(record, trace.context_);
+                    return;
+                }
             }
         }
+        catch (krabs::could_not_find_schema&) {}
```

Comment:
This exception should not be caught. How else will a caller know why the callback is not working?
This layer needs to bubble up errors so that the calling application can know something went wrong.

Reply:
And trace session will close without any chance to resume. In my case we can still handle event in default callback.